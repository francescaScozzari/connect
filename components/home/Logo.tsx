import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const BigLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="395"
      height="122"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M56.8333 5.38354H11.7031V52.7563c5.5773-25.1345 20.4454-40.9681 45.1302-47.37276ZM11.6992 70.4567v46.7523h46.126c-25.255-6.107-40.4903-20.8972-46.126-46.7523ZM67.5977 105.379c-2.1168-11.6491-6.8235-21.3173-14.7819-29.202-7.9306-7.8598-17.5191-12.6285-29.0853-14.5482 24.2424-5.2918 38.7294-19.6981 44.0619-44.1704 1.9027 11.8438 6.7846 21.4147 14.718 29.3579 7.9333 7.9432 17.5946 12.609 29.0996 14.6149-24.1538 5.1499-38.5908 19.431-44.0123 43.9478ZM123.505 69.9977v47.2113H77.4453c25.2547-6.107 40.4207-21.3562 46.0597-47.2113ZM78.3652 5.38354h45.1388V52.5699c-5.58-25.1345-20.457-40.7817-45.1388-47.18636ZM145.585 115.161c0 1.517-1.007 2.382-3.021 2.382h-3.652v-9.006h3.419c1.755 0 2.823.809 2.823 2.304 0 1.068-.551 1.763-1.371 2.058 1.16.242 1.802 1.016 1.802 2.265v-.003Zm-6.075-6.059v3.586h2.796c1.416 0 2.236-.604 2.236-1.8 0-1.197-.82-1.786-2.236-1.786h-2.796Zm5.477 6.006c0-1.285-.854-1.852-2.423-1.852h-3.054v3.717h3.054c1.569 0 2.423-.568 2.423-1.865ZM147.076 113.038c0-2.612 1.814-4.568 4.262-4.568 2.448 0 4.261 1.942 4.261 4.568 0 2.627-1.827 4.566-4.261 4.566s-4.262-1.956-4.262-4.566Zm7.914 0c0-2.278-1.569-3.964-3.652-3.964-2.084 0-3.664 1.686-3.664 3.964 0 2.279 1.569 3.962 3.664 3.962 2.094 0 3.652-1.683 3.652-3.962ZM157.094 113.038c0-2.612 1.813-4.568 4.261-4.568 2.448 0 4.262 1.942 4.262 4.568 0 2.627-1.828 4.566-4.262 4.566-2.434 0-4.261-1.956-4.261-4.566Zm7.914 0c0-2.278-1.569-3.964-3.653-3.964-2.083 0-3.663 1.686-3.663 3.964 0 2.279 1.569 3.962 3.663 3.962 2.095 0 3.653-1.683 3.653-3.962ZM166.949 116.383l.27-.487c.562.643 1.605 1.129 2.704 1.129 1.641 0 2.364-.784 2.364-1.761 0-2.701-5.115-1.082-5.115-4.398 0-1.272.89-2.393 2.901-2.393.901 0 1.839.309 2.47.81l-.222.528c-.679-.512-1.511-.759-2.248-.759-1.591 0-2.306.809-2.306 1.8 0 2.701 5.116 1.107 5.116 4.373 0 1.275-.924 2.382-2.949 2.382-1.218 0-2.387-.501-2.985-1.221v-.003ZM176.818 109.127h-2.949v-.593h6.498v.593h-2.951v8.413h-.595v-8.413h-.003ZM181.955 108.534h.598v9.006h-.598v-9.006ZM192.031 108.534v9.006h-.493l-5.583-7.824v7.824h-.598v-9.006h.504l5.572 7.824v-7.824h.598ZM200.83 113.075h.575v3.369c-.737.771-1.802 1.157-2.926 1.157-2.47 0-4.284-1.942-4.284-4.565 0-2.624 1.814-4.569 4.298-4.569 1.182 0 2.223.412 2.949 1.236l-.376.436c-.715-.77-1.569-1.068-2.565-1.068-2.13 0-3.71 1.7-3.71 3.965 0 2.264 1.58 3.962 3.71 3.962.865 0 1.661-.231 2.329-.821v-3.102ZM206.93 108.534h.598v9.006h-.598v-9.006ZM217.005 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.504l5.571 7.824v-7.824h.598ZM225.806 113.075h.576v3.369c-.737.771-1.803 1.157-2.926 1.157-2.47 0-4.284-1.942-4.284-4.565 0-2.624 1.814-4.569 4.298-4.569 1.182 0 2.222.412 2.948 1.236l-.375.436c-.715-.77-1.569-1.068-2.565-1.068-2.131 0-3.711 1.7-3.711 3.965 0 2.264 1.58 3.962 3.711 3.962.865 0 1.661-.231 2.328-.821v-3.102ZM234.628 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.367v.578h-4.367v3.692h5.057ZM243.273 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.503l5.572 7.824v-7.824h.598ZM246.09 108.534h.598v9.006h-.598v-9.006ZM249.442 113.717v-5.183h.596v5.158c0 2.251.984 3.306 2.692 3.306 1.708 0 2.682-1.055 2.682-3.306v-5.158h.595v5.183c0 2.574-1.252 3.884-3.291 3.884-2.039 0-3.277-1.31-3.277-3.884h.003ZM266.518 117.54l-.011-7.707-3.429 6.511h-.293l-3.429-6.486v7.682h-.585v-9.006h.504l3.663 6.958 3.664-6.958h.492l.014 9.006h-.59ZM273.459 109.127v3.845h4.368v.592h-4.368v3.976h-.598v-9.006h5.48v.593h-4.882ZM279.504 113.038c0-2.612 1.814-4.568 4.261-4.568 2.448 0 4.262 1.942 4.262 4.568 0 2.627-1.828 4.566-4.262 4.566-2.434 0-4.261-1.956-4.261-4.566Zm7.914 0c0-2.278-1.569-3.964-3.653-3.964-2.083 0-3.663 1.686-3.663 3.964 0 2.279 1.569 3.962 3.663 3.962 2.095 0 3.653-1.683 3.653-3.962ZM295.882 117.54l-1.956-3.01c-.258.039-.514.064-.795.064h-2.34v2.946h-.598v-9.006h2.938c2.061 0 3.302 1.132 3.302 3.035 0 1.428-.704 2.418-1.92 2.819l2.048 3.152h-.679Zm-2.751-3.525c1.755 0 2.692-.901 2.692-2.443 0-1.541-.937-2.445-2.692-2.445h-2.34v4.888h2.34ZM307.179 116.947v.593h-5.656v-9.006h5.48v.593h-4.881v3.55h4.367v.578h-4.367v3.692h5.057ZM314.841 117.54l-2.821-4.168-2.809 4.168h-.701l3.149-4.643-2.927-4.363h.701l2.612 3.873 2.61-3.873h.667l-2.926 4.349 3.149 4.657h-.704ZM316.264 113.038c0-2.626 1.813-4.568 4.25-4.568 1.157 0 2.2.412 2.915 1.249l-.375.423c-.701-.759-1.558-1.068-2.518-1.068-2.097 0-3.677 1.7-3.677 3.964 0 2.265 1.58 3.962 3.677 3.962.96 0 1.814-.32 2.518-1.079l.375.426c-.715.834-1.755 1.26-2.915 1.26-2.434 0-4.25-1.942-4.25-4.566v-.003ZM331.052 116.947v.593h-5.656v-9.006h5.48v.593h-4.881v3.55h4.367v.578h-4.367v3.692h5.057ZM333.027 108.534h.598v8.413h4.718v.593h-5.316v-9.006ZM339.967 108.534h.598v8.413h4.718v.593h-5.316v-9.006ZM352.563 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.368v.578h-4.368v3.692h5.057ZM361.21 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.504l5.571 7.824v-7.824h.598ZM363.373 113.038c0-2.626 1.814-4.568 4.25-4.568 1.158 0 2.201.412 2.916 1.249l-.376.423c-.701-.759-1.558-1.068-2.517-1.068-2.098 0-3.678 1.7-3.678 3.964 0 2.265 1.58 3.962 3.678 3.962.959 0 1.813-.32 2.517-1.079l.376.426c-.715.834-1.756 1.26-2.916 1.26-2.434 0-4.25-1.942-4.25-4.566v-.003ZM378.159 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.367v.578h-4.367v3.692h5.057ZM193.275 49.287c7.992 4.165 13.008 12.5032 13.008 22.6778 0 15.1046-11.859 26.5451-27.514 26.5451h-39.091V5.29187h38.309c15.038 0 26.012 10.08553 26.012 23.85743 0 8.33-4.245 15.8225-10.724 20.1377ZM181.82 33.4033c0-4.8382-3.015-8.1157-7.902-8.1157h-11.809v16.2315h11.809c4.887 0 7.902-3.2775 7.902-8.1158Zm1.349 35.5094c0-4.9495-3.307-8.4384-8.367-8.4384h-12.693v16.952h12.693c5.06 0 8.367-3.4861 8.367-8.5164v.0028ZM217.033 5.28894h23.675V98.507h-23.675V5.28894ZM313.862 82.5982h-7.279v15.9115h-20.704V82.5982h-38.68v-4.3319l25.812-68.3172h21.396l-19.956 52.8036h12.885l4.726-19.5033h14.518v19.5033h7.28v19.8427l.002.0028ZM378.16 76.2326V98.507h-55.295V5.28894h54.268V27.4882h-31.004v13.7219h27.855v21.0641h-27.855v13.9556h32.031v.0028Z"
        fill="#000"
      />
    </svg>
  )
}

const SmallLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="211"
      height="65"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g fill="#000">
        <path d="M30.3594 2.86829H6.25195V28.1079C9.23121 14.7165 17.1734 6.2806 30.3594 2.86829ZM6.25 37.5385v24.909h24.6395C17.3988 59.1938 9.26046 51.3137 6.25 37.5385ZM36.1087 56.1446c-1.1308-6.2065-3.645-11.3576-7.8962-15.5585-4.2363-4.1875-9.3583-6.7283-15.5367-7.7511 12.9497-2.8193 20.6884-10.4948 23.5369-23.53336 1.0164 6.31026 3.6241 11.40946 7.862 15.64146 4.2378 4.2321 9.3984 6.7179 15.5441 7.7867-12.9022 2.7438-20.6141 10.3526-23.5101 23.4148ZM65.9729 37.2938v25.1536H41.3691c13.4907-3.2537 21.5919-11.3783 24.6038-25.1536ZM41.8613 2.86829h24.112V28.0085C62.9925 14.6172 55.0459 6.2806 41.8613 2.86829ZM77.7678 61.3564c0 .8079-.5379 1.2689-1.6137 1.2689h-1.951V57.827h1.8262c.9376 0 1.5082.4314 1.5082 1.2274 0 .5692-.2942.9398-.7325 1.0969.6196.129.9628.5411.9628 1.2066v-.0015Zm-3.2452-3.2285v1.9108h1.4933c.7564 0 1.1947-.3217 1.1947-.9591s-.4383-.9517-1.1947-.9517h-1.4933Zm2.9258 3.2004c0-.6849-.4562-.9872-1.2943-.9872h-1.6315v1.9803h1.6315c.8381 0 1.2943-.3024 1.2943-.9931ZM78.5645 60.2254c0-1.3919.9688-2.434 2.2764-2.434 1.3076 0 2.2764 1.0346 2.2764 2.434 0 1.3993-.9762 2.4325-2.2764 2.4325-1.3002 0-2.2764-1.0421-2.2764-2.4325Zm4.2274 0c0-1.2141-.8381-2.1124-1.951-2.1124-1.113 0-1.957.8983-1.957 2.1124 0 1.214.8381 2.1108 1.957 2.1108s1.951-.8968 1.951-2.1108ZM83.916 60.2254c0-1.3919.9688-2.434 2.2764-2.434 1.3076 0 2.2765 1.0346 2.2765 2.434 0 1.3993-.9763 2.4325-2.2765 2.4325-1.3001 0-2.2764-1.0421-2.2764-2.4325Zm4.2274 0c0-1.2141-.838-2.1124-1.951-2.1124-1.1129 0-1.9569.8983-1.9569 2.1124 0 1.214.838 2.1108 1.9569 2.1108s1.951-.8968 1.951-2.1108ZM89.1797 62.0071l.1441-.2594c.3002.3424.8574.6018 1.4443.6018.8767 0 1.2631-.418 1.2631-.9383 0-1.4393-2.7326-.5766-2.7326-2.3436 0-.6774.4755-1.2748 1.5498-1.2748.4814 0 .9822.1646 1.3195.4314l-.1189.2816c-.3626-.2727-.8069-.4046-1.2006-.4046-.85 0-1.2319.4313-1.2319.959 0 1.4394 2.7326.59 2.7326 2.3303 0 .6789-.4933 1.2688-1.575 1.2688-.6508 0-1.2749-.2668-1.5944-.6507v-.0015ZM94.452 58.1412h-1.575v-.3158h3.471v.3158h-1.5765v4.4825h-.318v-4.4825h-.0015ZM97.1953 57.8254h.3195v4.7983h-.3195v-4.7983ZM102.578 57.8254v4.7983h-.263l-2.9819-4.1683v4.1683h-.3194v-4.7983h.2689l2.9764 4.1683v-4.1683h.319ZM107.278 60.2447h.308v1.795c-.394.4107-.963.6167-1.563.6167-1.32 0-2.289-1.0347-2.289-2.4325 0-1.3978.969-2.434 2.296-2.434.632 0 1.187.2194 1.575.6582l-.2.2327c-.382-.4106-.838-.5692-1.37-.5692-1.139 0-1.983.9057-1.983 2.1123s.844 2.1108 1.983 2.1108c.462 0 .887-.123 1.243-.4373v-1.6527ZM110.537 57.8254h.32v4.7983h-.32v-4.7983ZM115.918 57.8254v4.7983h-.263l-2.982-4.1683v4.1683h-.319v-4.7983h.268l2.977 4.1683v-4.1683h.319ZM120.62 60.2447h.308v1.795c-.394.4107-.963.6167-1.564.6167-1.319 0-2.288-1.0347-2.288-2.4325 0-1.3978.969-2.434 2.296-2.434.631 0 1.187.2194 1.575.6582l-.201.2327c-.381-.4106-.838-.5692-1.37-.5692-1.138 0-1.982.9057-1.982 2.1123s.844 2.1108 1.982 2.1108c.462 0 .887-.123 1.244-.4373v-1.6527ZM125.331 62.308v.3157h-3.02v-4.7983h2.927v.3158h-2.608v1.8914h2.333v.3083h-2.333v1.9671h2.701ZM129.949 57.8254v4.7983h-.263l-2.982-4.1683v4.1683h-.319v-4.7983h.269l2.976 4.1683v-4.1683h.319ZM131.455 57.8254h.32v4.7983h-.32v-4.7983ZM133.246 60.587v-2.7616h.318v2.7483c0 1.1992.526 1.761 1.438 1.761.912 0 1.432-.5618 1.432-1.761v-2.7483h.318v2.7616c0 1.3712-.668 2.0693-1.757 2.0693-1.09 0-1.751-.6981-1.751-2.0693h.002ZM142.369 62.6237l-.006-4.106-1.832 3.4686h-.156l-1.832-3.4553v4.0927h-.313v-4.7983h.269l1.957 3.7073 1.957-3.7073h.263l.008 4.7983h-.315ZM146.075 58.1412v2.0485h2.333v.3158h-2.333v2.1182h-.319v-4.7983h2.927v.3158h-2.608ZM149.305 60.2254c0-1.3919.969-2.434 2.276-2.434 1.308 0 2.277 1.0346 2.277 2.434 0 1.3993-.977 2.4325-2.277 2.4325-1.3 0-2.276-1.0421-2.276-2.4325Zm4.227 0c0-1.2141-.838-2.1124-1.951-2.1124s-1.957.8983-1.957 2.1124c0 1.214.838 2.1108 1.957 2.1108 1.119 0 1.951-.8968 1.951-2.1108ZM158.052 62.6237l-1.044-1.6038c-.138.0207-.275.034-.425.034h-1.25v1.5698h-.319v-4.7983h1.569c1.101 0 1.764.6033 1.764 1.6173 0 .7604-.376 1.2881-1.026 1.5016l1.094 1.6794h-.363Zm-1.469-1.8781c.937 0 1.438-.4803 1.438-1.3015 0-.8212-.501-1.3029-1.438-1.3029h-1.25v2.6044h1.25ZM164.087 62.308v.3157h-3.021v-4.7983h2.928v.3158h-2.608v1.8914h2.333v.3083h-2.333v1.9671h2.701ZM168.181 62.6237l-1.507-2.2205-1.501 2.2205h-.374l1.682-2.474-1.563-2.3243h.374l1.395 2.0634 1.394-2.0634h.357l-1.563 2.3169 1.682 2.4814h-.376ZM168.941 60.2254c0-1.3994.969-2.434 2.271-2.434.618 0 1.175.2194 1.557.6655l-.2.2254c-.375-.4047-.833-.5693-1.345-.5693-1.121 0-1.965.9057-1.965 2.1124 0 1.2066.844 2.1108 1.965 2.1108.512 0 .969-.1705 1.345-.5751l.2.2267c-.382.4447-.937.6715-1.557.6715-1.3 0-2.271-1.0346-2.271-2.4325v-.0014ZM176.839 62.308v.3157h-3.021v-4.7983h2.928v.3158h-2.608v1.8914h2.333v.3083h-2.333v1.9671h2.701ZM177.895 57.8254h.319v4.4826h2.52v.3157h-2.839v-4.7983ZM181.602 57.8254h.319v4.4826h2.52v.3157h-2.839v-4.7983ZM188.329 62.308v.3157h-3.02v-4.7983h2.927v.3158h-2.608v1.8914h2.333v.3083h-2.333v1.9671h2.701ZM192.949 57.8254v4.7983h-.263l-2.982-4.1683v4.1683h-.319v-4.7983h.269l2.976 4.1683v-4.1683h.319ZM194.105 60.2254c0-1.3994.969-2.434 2.271-2.434.618 0 1.175.2194 1.557.6655l-.2.2254c-.375-.4047-.833-.5693-1.345-.5693-1.121 0-1.965.9057-1.965 2.1124 0 1.2066.844 2.1108 1.965 2.1108.512 0 .969-.1705 1.345-.5751l.2.2267c-.382.4447-.937.6715-1.557.6715-1.3 0-2.271-1.0346-2.271-2.4325v-.0014ZM202.003 62.308v.3157h-3.021v-4.7983h2.928v.3158h-2.608v1.8914h2.333v.3083h-2.333v1.9671h2.701ZM103.242 26.2596c4.269 2.219 6.948 6.6616 6.948 12.0824 0 8.0476-6.334 14.1429-14.6971 14.1429H74.6113V2.81958h20.464c8.0327 0 13.8947 5.37343 13.8947 12.71092 0 4.4381-2.267 8.43-5.728 10.7291Zm-6.1191-8.4626c0-2.5778-1.6107-4.3239-4.2214-4.3239h-6.3077v8.6478h6.3077c2.6107 0 4.2214-1.7461 4.2214-4.3239Zm.7207 18.9189c0-2.637-1.7667-4.4959-4.4696-4.4959h-6.7802v9.0319h6.7802c2.7029 0 4.4696-1.8574 4.4696-4.5375v.0015ZM115.934 2.81787h12.646V52.4832h-12.646V2.81787ZM167.657 44.0073h-3.889v8.4774h-11.059v-8.4774h-20.662v-2.308l13.788-36.39852h11.429l-10.66 28.13302h6.883l2.525-10.3911h7.755v10.3911h3.888v10.572l.002.0015ZM202.004 40.6157v11.8675h-29.537V2.81787h28.989V14.6454h-16.562v7.3108h14.879v11.2227h-14.879v7.4353h17.11v.0015Z" />
      </g>
    </svg>
  )
}

export { BigLogo, SmallLogo }
