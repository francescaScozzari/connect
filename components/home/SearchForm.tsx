import React, { useRef } from 'react'
import { styled } from 'styled-components'
import { useForm } from 'react-hook-form'
import type { SubmitHandler } from 'react-hook-form'

import { Text } from '@/components/commons/Typography'
import {
  IconArrowSubmit,
  IconCrossReset,
  IconSearchVariant
} from '@/components/commons/Icons'
import { useDynamicHeight } from '@/hooks'

type FormValues = {
  givenSentence: string
}

type Props = {
  handleSubmit: (body: FormValues) => void
}

const SearchForm = ({ handleSubmit }: Props) => {
  const {
    register,
    reset,
    handleSubmit: handleHookFormSubmit,
    formState: { errors, isSubmitting, isValid }
  } = useForm<FormValues>({ mode: 'onChange' })

  const textAreaRef = useRef<HTMLTextAreaElement | null>(null)
  const heightKeeperRef = useRef<HTMLDivElement | null>(null)
  const { ref, ...rest } = register('givenSentence', {
    required: 'A search prompt is required',
    maxLength: 1024
  })

  useDynamicHeight(textAreaRef?.current, heightKeeperRef?.current)

  const onSubmit: SubmitHandler<FormValues> = async body => {
    handleSubmit({ ...body })
  }

  return (
    <Form onSubmit={handleHookFormSubmit(onSubmit)} role="search">
      <Wrapper>
        <HeightKeeper ref={heightKeeperRef} />
        <Input
          id="q"
          placeholder="Search here to find your research team"
          role="textarea"
          ref={e => {
            ref(e)
            textAreaRef.current = e
          }}
          {...rest}
        />
        {errors.givenSentence?.type === 'maxLength' ? (
          <Text.NormalBold role="alert">Search prompt too long</Text.NormalBold>
        ) : null}
        <Placeholder $isFilled={textAreaRef?.current?.value} title="search" />
        <ClearButton
          type="button"
          role="reset"
          $isFilled={textAreaRef?.current?.value}
          onClick={() => {
            reset()
            textAreaRef?.current?.dispatchEvent(new Event('input'))
          }}
        >
          <>
            <IconCrossReset title="reset" />
          </>
        </ClearButton>
        <SubmitButton
          type="submit"
          role="submit"
          disabled={isSubmitting || !isValid}
        >
          <IconArrowSubmit title="submit" />
        </SubmitButton>
      </Wrapper>
    </Form>
  )
}

const Placeholder = styled(IconSearchVariant)<{
  $isFilled: string | undefined
}>`
  display: ${({ $isFilled }) => ($isFilled?.length ? 'none' : 'block')};
  position: absolute;
  top: calc(6em / 2);
  left: 5%;
  transform: translate(-50%, -50%);

  @media (max-width: 1280px) {
    top: calc(6em / 2);
  }

  @media (max-width: 768px) {
    top: calc(6.125em / 2);
    left: 7%;
  }
`

const HeightKeeper = styled.div`
  display: none;
`

const ClearButton = styled.button<{ $isFilled: string | undefined }>`
  all: unset;
  display: ${({ $isFilled }) => ($isFilled?.length ? 'block' : 'none')};
  position: absolute;
  top: calc(6em / 2);
  left: 5%;
  transform: translate(-50%, -50%);

  @media (max-width: 1280px) {
    top: calc(6em / 2);
  }

  @media (max-width: 768px) {
    top: calc(6.125em / 2);
    left: 7%;
  }
`

const SubmitButton = styled.button`
  width: 4.5em;
  height: 4.5em;
  border: 2px solid black;
  border-radius: 50%;
  position: absolute;
  top: calc(5.75em / 2);
  right: 0;
  transform: translate(-50%, -50%);

  &:disabled {
    opacity: 0.3;
  }

  @media (max-width: 1280px) {
    top: calc(5.5em / 2);
  }

  @media (max-width: 768px) {
    width: 4em;
    height: 4em;
    top: calc(5.5em / 2);
  }
`

const Wrapper = styled.div`
  display: flex;
  position: relative;
  flex-direction: column;
  gap: 2em;
  align-items: center;
`

const Form = styled.form`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
  p {
    margin-left: 0.625em;
    color: ${({ theme }) => theme.colors.status.error};
  }
`

const Input = styled.textarea`
  position: relative;
  flex-grow: 1;
  resize: none;
  overflow: hidden;
  border: 2px solid black;
  border-radius: 3.5em;
  box-sizing: border-box;
  width: 52em;
  padding: 1.75em 6.25em 0.75em 4.625em;
  line-height: 1.375;
  font-size: 1.125rem;

  &::placeholder {
    opacity: 1;
  }

  @media (max-width: 1280px) {
    width: 42em;
    padding: 1.625em 6.25em 0.75em 4.625em;
  }

  @media (max-width: 768px) {
    font-size: 1em;
    width: 36em;
    padding: 2em 6.25em 0.75em 4.625em;
  }
`

export { SearchForm }
