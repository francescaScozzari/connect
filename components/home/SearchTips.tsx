import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { IconCopy, IconTarget } from '../commons/Icons'

const SearchTips = () => {
  return (
    <OuterWrapper>
      <Text.NormalBold>
        Search tips to get the best research team suggestion
      </Text.NormalBold>
      <InnerWrapper>
        <SubWrapper>
          <IconCopy title={'paper'} />
          <Text.Normal>
            <Highlight>Start from the the research call</Highlight>
            <br /> If you are working or will have to work on a research paper
            for a call, you might report in the search field the title or
            objectives of that
          </Text.Normal>
        </SubWrapper>

        <SubWrapper>
          <IconTarget title={'target'} />
          <Text.Normal>
            <Highlight>More detail for greater match</Highlight>
            <br /> Enter as much information, don't limit yourself to a single
            word, that way we can narrow it down and suggest the researchers
            best suited to your need
          </Text.Normal>
        </SubWrapper>
      </InnerWrapper>
    </OuterWrapper>
  )
}

const OuterWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 2em;
  max-width: 53em;
  background-color: #f5f5f5;
  padding: 4em 2em;

  p {
    align-self: center;
  }

  @media (max-width: 1280px) {
    max-width: 46em;
  }

  @media (max-width: 768px) {
    max-width: 37.5em;
  }
`
const InnerWrapper = styled.div`
  display: flex;
  gap: 2em;
`
const SubWrapper = styled.div`
  display: flex;
  gap: 0.5em;
  align-items: start;

  svg {
    flex-shrink: 0;
  }
`
const Highlight = styled.span`
  display: inline;
  font-size: ${({ theme }) => theme.texts.normalBold.fontSize};
  font-weight: ${({ theme }) => theme.texts.normalBold.fontWeight};
`

export { SearchTips }
