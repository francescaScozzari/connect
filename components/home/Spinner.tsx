import React from 'react'
import styled from 'styled-components'
import GridLoader from 'react-spinners/GridLoader'

import { H2 } from '@/components/commons/Typography'

export const Spinner = ({ spinnerLoading }: { spinnerLoading: boolean }) => {
  return (
    <FloatingCointainer>
      <Wrapper>
        <GridLoader color={'#CE2F7C'} size={40} loading={spinnerLoading} />
        <H2.Normal color="white">We are searching...</H2.Normal>
      </Wrapper>
    </FloatingCointainer>
  )
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 1em;
  position: absolute;
  z-index: 1000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

const FloatingCointainer = styled.div`
  background-color: ${({ theme }) => theme.colors.primary[0]};
  position: absolute;
  z-index: 500;
  height: calc(100vh + 80px);
  width: 100%;
`
