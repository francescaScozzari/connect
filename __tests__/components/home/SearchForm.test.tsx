import React from 'react'
import { expect } from '@jest/globals'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'

import { SearchForm } from '@/components/home/SearchForm'
import { renderWithWrappers } from '@/__tests__/functions'

jest.mock('next/router', () => require('next-router-mock'))

describe('<SearchForm />', () => {
  it('SearchForm renders correctly', () => {
    const container = renderWithWrappers(<SearchForm />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display required error when input is empty', async () => {
    renderWithWrappers(<SearchForm />)
    fireEvent.submit(screen.getByRole('button'))
    expect(await screen.findByText('A search prompt is required')).toBeDefined()
  })

  it('should display matching error when input is too long', async () => {
    renderWithWrappers(<SearchForm />)
    fireEvent.input(screen.getByRole('textbox'), {
      target: {
        value: 'x'.repeat(2 * 1024)
      }
    })

    fireEvent.submit(screen.getByRole('button'))
    expect(await screen.findByText('Search prompt too long')).toBeDefined()
  })
})
