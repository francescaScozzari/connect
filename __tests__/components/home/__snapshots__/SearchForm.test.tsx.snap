// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> SearchForm renders correctly 1`] = `
.c2 {
  display: none;
}

.c4 {
  all: unset;
  display: none;
  position: absolute;
  top: calc(6em / 2);
  left: 5%;
  transform: translate(-50%, -50%);
}

.c5 {
  width: 4.5em;
  height: 4.5em;
  border: 2px solid black;
  border-radius: 50%;
  position: absolute;
  top: calc(5.75em / 2);
  right: 0;
  transform: translate(-50%, -50%);
}

.c5:disabled {
  opacity: 0.3;
}

.c1 {
  display: flex;
  position: relative;
  flex-direction: column;
  gap: 2em;
  align-items: center;
}

.c0 {
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
}

.c0 p {
  margin-left: 0.625em;
  color: #BA1B23;
}

.c3 {
  position: relative;
  flex-grow: 1;
  resize: none;
  overflow: hidden;
  border: 2px solid black;
  border-radius: 3.5em;
  box-sizing: border-box;
  width: 52em;
  padding: 1.75em 6.25em 0.75em 4.625em;
  line-height: 1.375;
  font-size: 1.125rem;
}

.c3::placeholder {
  opacity: 1;
  background: url("data:image/svg+xml,%3Csvg height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:none;stroke:%23000;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px;%7D%3C/style%3E%3C/defs%3E%3Ctitle/%3E%3Cg id='_21.search'%3E%3Ccircle class='cls-1' cx='9' cy='9' r='8'/%3E%3Cline class='cls-1' x1='15' x2='23' y1='15' y2='23'/%3E%3C/g%3E%3C/svg%3E") no-repeat 2em center;
}

@media (max-width: 1280px) {
  .c4 {
    top: calc(6em / 2);
  }
}

@media (max-width: 768px) {
  .c4 {
    top: calc(6.125em / 2);
    left: 7%;
  }
}

@media (max-width: 1280px) {
  .c5 {
    top: calc(5.5em / 2);
  }
}

@media (max-width: 768px) {
  .c5 {
    width: 4em;
    height: 4em;
    top: calc(5.5em / 2);
  }
}

@media (max-width: 1280px) {
  .c3 {
    width: 42em;
    padding: 1.625em 6.25em 0.75em 4.625em;
  }
}

@media (max-width: 768px) {
  .c3 {
    font-size: 1em;
    width: 36em;
    padding: 2em 6.25em 0.75em 4.625em;
  }
}

<form
  class="c0"
  role="search"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
    <textarea
      class="c3"
      id="q"
      name="q"
      placeholder="Search here to find your research team"
      role="textarea"
    />
    <button
      class="c4"
      role="reset"
      type="button"
    >
      <div>
        <svg
          fill="none"
          height="31"
          width="31"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            reset
          </title>
          <g>
            <path
              d="m24.5417 8.27962-1.8212-1.82125L15.5 13.6788 8.27962 6.45837 6.45837 8.27962 13.6788 15.5l-7.22043 7.2205 1.82125 1.8212L15.5 17.3213l7.2205 7.2204 1.8212-1.8212L17.3213 15.5l7.2204-7.22038Z"
              fill="#000"
            />
          </g>
        </svg>
      </div>
    </button>
    <button
      class="c5"
      disabled=""
      role="submit"
      type="submit"
    >
      <svg
        fill="none"
        height="45"
        viewBox="0 0 45 45"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          submit
        </title>
        <g
          id="Arrow forward"
        >
          <path
            d="M22.5 7.5L19.8563 10.1437L30.3188 20.625H7.5V24.375H30.3188L19.8563 34.8563L22.5 37.5L37.5 22.5L22.5 7.5Z"
            fill="black"
            id="Vector"
          />
        </g>
      </svg>
    </button>
  </div>
</form>
`;
